gensim/models/word2vec.py

```diff
@@ -910,12 +915,14 @@ def _do_train_epoch(
         if self.sg:
             examples, tally, raw_tally = train_epoch_sg(
                 self, corpus_file, offset, cython_vocab, cur_epoch,
-                total_examples, total_words, work, neu1, self.compute_loss,
+                total_examples, total_words, work, neu1,
+                self.compute_loss, self.reduced_windows,
```

Comment:
In general, if the parameter `window` wasn't being passed-along in a widened method-signature - because the inner implementation method was able to read it from elsewhere, then this new closely-related `reduced_windows` parameter probably should be consulted the same way. So: I don't think any method that didn't already have `window` in its signature shouild be adding `reduced_windows`.

Reply:
I agree for all "backend" methods and functions and have corrected them.
I have a slight hesitation regarding the `train` method (see discussion below).